Cairo Desktop/Cairo Desktop/MenuBar.xaml.cs

```diff
@@ -862,8 +862,29 @@ private void SysSleep(object sender, RoutedEventArgs e)
 
         private void InitCairoSettingsWindow(object sender, RoutedEventArgs e)
         {
-            CairoSettingsWindow window = new CairoSettingsWindow();
-            window.Show();
+            string windowName = "CairoSettingsWindow";
+
+            if (IsOpenWindow(windowName).Equals(false))
```

Comment:
@Brummell67 You have the right idea but I think that we can optimize this slightly by storing a reference to CairoSettingsWindow outside of the InitCairoSettingsWindow event. This will allow you to test it without iterating over the Application.Current.Windows collection. The key is to clear the variable and cleanup when the CairoSettingsWindow is Closed

Reply:
If I stored bool variable in MenuBar.xaml.cs, is there a way for intercept InitCairoSettingsWindow closing event, without using global variable, then keep decoupling?